packages/backend/src/services/OrganizationService/OrganizationService.ts

```diff
@@ -752,4 +778,157 @@ export class OrganizationService extends BaseService {
+                    `Initial setup: There is already an organization, we skip this initial setup`,
+                );
+                // There is already an organization, we skip this step
+                return;
+            }
+            const hasAnyProjects = await this.projectModel.hasAnyProjects();
+
+            if (hasAnyProjects) {
+                // This should not happen, since we can't have projects without orgs
+                throw new UnexpectedServerError(
+                    `Initial setup: There is already a project, we skip this initial setup`,
+                );
+            }
+
+            // No organization and no projects, we can create a new one
+            // using the initial setup config
+            this.logger.debug(
+                `Initial setup: Creating organization "${setup.organization.name}"`,
+            );
+
+            const organization = await this.organizationModel.create({
+                name: setup.organization.name,
+            });
+            const { organizationUuid } = organization;
+
+            this.logger.info(
+                `Initial setup: Organization "${organizationUuid}" created`,
+            );
+
+            if (setup.organization.emailDomain) {
+                const emailDomains = [setup.organization.emailDomain];
+                // Validates input
+                const error = validateOrganizationEmailDomains(emailDomains);
+                if (error) {
+                    throw new ParameterError(error);
+                }
+                const allowedDomains: AllowedEmailDomains = {
+                    organizationUuid,
+                    emailDomains,
+                    role: OrganizationMemberRole.VIEWER,
+                    projects: [],
+                };
+                await this.organizationAllowedEmailDomainsModel.upsertAllowedEmailDomains(
+                    allowedDomains,
+                );
+
+                this.logger.info(
+                    `Initial setup: Whitelisted domain "${setup.organization.emailDomain}"`,
+                );
+            } else {
+                this.logger.info(
+                    `Initial setup: No whitelisted domain, skipping`,
+                );
+            }
+
+            this.logger.debug(
+                `Initial setup: Creating admin user with email "${setup.organization.admin.email}"`,
+            );
+
+            // We need `AUTH_ENABLE_OIDC_TO_EMAIL_LINKING=true`
+            // So the user can login using SSO for the pending user
+            const { email, name: adminName } = setup.organization.admin;
+            const user = await this.userModel.createPendingUser(
+                organizationUuid,
+                {
+                    firstName: adminName.split(' ')[0],
+                    lastName: adminName.split(' ').slice(1).join(' '),
+                    email,
+                    role: OrganizationMemberRole.ADMIN,
+                    password: undefined,
```

Comment:
with password is much simpler to explain than OIDC linking and doesn't depend on it

Reply:
Or make this optional?